server/common/config/external_config.py

```diff
@@ -0,0 +1,103 @@
+import os
+
+from server.common.config.base_config import BaseConfig
+from server.common.errors import ConfigurationError
+from server.common.config import get_secret_key
+from server.common.errors import SecretKeyRetrievalError
+
+
+class ExternalConfig(BaseConfig):
+    """Manages the config attribute associated with external configuration sources, such as
+    environment variables or the AWS Secrets Manager."""
+
+    def __init__(self, app_config, default_config):
+        super().__init__(app_config, default_config)
+        try:
+            self.environment = default_config["environment"]
+            self.aws_secrets_manager__region = default_config["aws_secrets_manager"]["region"]
+            self.aws_secrets_manager__secrets = default_config["aws_secrets_manager"]["secrets"]
+
+        except KeyError as e:
+            raise ConfigurationError(f"Unexpected config: {str(e)}")
+
+    @staticmethod
+    def convert_value(value: str):
+        """convert the external value (a string), to the most appropriate type"""
+        if value == "true":
+            return True
+        if value == "false":
+            return False
+        if value == "null":
+            return None
+        try:
+            return eval(value)
```

Comment:
What case is this catching?

Reply:
The eval is converting to ints or double or dictionaries, etc.
E.g. "1" will convert to int(1).
If the value doesn't eval, like "localhost:5005", then it's caught and we assume the wanted a string.

Reply:
Actually, I suppose someone could construct a string that failed in a different way than NameError or SyntaxError.  I'll just catch any exception here.